upgrade/steps.go

```diff
@@ -556,110 +558,112 @@ func MajorVersionBump(ctx context.Context, goMod *GoMod, target *UpstreamUpgrade
 
 	// Replace s in file, where {} is interpolated into the old and new provider
 	// component of the path.
-	replaceInFile := func(desc, path, s string) step.Step {
-		return UpdateFile(desc+" in "+path, path, func(src []byte) ([]byte, error) {
+	replaceInFile := func(desc, path, s string) {
```

Comment:
`replaceInFile` does more than that: it also constructs the replace statement without duplication. We could inline it, but I think `replaceInFile` is easer to read then the inlined version. To inline `replaceInFile`, we would go from:

```go
		replaceInFile("Update PROVIDER_PATH", "Makefile",
			"PROVIDER_PATH := provider{}")
```

to

```go
		updateFile(ctx, "Makefile", func(ctx context.Context, src string) string {
			stepv2.SetLabel(ctx, "Update PROVIDER_PATH")
			return strings.ReplaceAll(src,
				"PROVIDER_PATH := provider"+prev,
				"PROVIDER_PATH := provider"+next)
		})
```

Reply:
I feel the opposite, because now I can immediately see what information `updateFile` is being updated _with_. This is incredibly useful for someone who is just reading the code, trying to see what happens when.

to elaborate: when first I read `replaceInFile`, below, I saw that there is _maybe_ some updating going on? I was very confused first reading this because I wasn't sure where or why the `provider{}` string was being set, and where the replace strings were.

If you want to avoid deduplication, we could maybe abstract the anonymous func inside `updateFile` as a non-inline helper function?

Reply:
I have moved `replaceInFile` to its own function. I want to avoid instantiating it because it will create a large number of string copies that *must* be identical, and it's easy to typo them.